types/aws-lambda/handler.d.ts

```diff
@@ -170,3 +176,36 @@ export interface ClientContextEnv {
  *   Pass `null` or `undefined` for the `error` parameter to use this parameter.
  */
 export type Callback<TResult = any> = (error?: Error | string | null, result?: TResult) => void;
+
+export type StreamHandler = (
+  event: APIGatewayProxyEvent,
```

Comment:
I don't think the event has to be from API Gateway. In fact, API Gateway doesn't support streaming the response to reduce TTFB, IIRC.

This should be generic in the event like Handler, and ideally every trigger would have a streaming alias too, but that's probably a bit too much to ask for.